src/state/RedisStateKeyValue.cpp

```diff
@@ -18,12 +18,17 @@ namespace state {
         return redis.strlen(actualKey);
     }
 
+    // TODO - the remote locking here is quite primitive since we ignore the fact threads can run
+    // on the same machine. Redis is also aware of scheduling and so we could optimise this.
```

Comment:
Agreed. Would require some careful thought on how to do without introducing deadlock potential. Let's see how long `RedisStateKeyValue` survives, it may be killed soon anyway.